rules/prefer-number-is-integer.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+const {methodCallSelector} = require('./selectors/index.js');
+const isSameReference = require('./utils/is-same-reference.js');
+
+const MESSAGE_ID = 'preferNumberIsInteger';
+const MESSAGE_ID_SUGGEST = 'preferNumberIsIntegerSuggestion';
+const messages = {
+	[MESSAGE_ID]: 'Replace `{{original}}` with `Number.isInteger({{variable}})`.',
+	[MESSAGE_ID_SUGGEST]: 'Prefer `Number.isInteger()` for integer checks.',
+};
+
+const equalsSelector = ':matches([operator="==="],[operator="=="])';
+
+// `value % 1 === 0`
+const modulo1Selector = [
+	'BinaryExpression',
+	'[left.type="BinaryExpression"]',
+	'[left.operator="%"]',
+	'[left.right.value=1]',
+	equalsSelector,
+	'[right.value="0"]',
+].join('');
+
+// `(value ^ 0) === value`
+// `(value | 0) === value`
+const mathOperatorSelector = [
+	'BinaryExpression',
+	'[left.type="BinaryExpression"]',
+	`:matches(${['^', '|'].map(operator => `[left.operator="${operator}"]`).join(',')})`,
+	'[left.right.value=0]',
+	equalsSelector,
+].join('');
+
+// `Number.parseInt(value,10) === value`
+const numberParseIntSelector = [
+	'BinaryExpression',
+	'[left.type="CallExpression"]',
+	'[left.callee.type="MemberExpression"]',
+	'[left.callee.object.name="Number"]',
+	'[left.callee.property.name="parseInt"]',
+	'[left.arguments.1.value=10]',
+	equalsSelector,
+].join('');
+
+// `_.isInteger(value)`
+const lodashIsIntegerSelector = [
+	methodCallSelector({method: 'isInteger', objects: ['_', 'lodash', 'underscore']}),
+].join('');
+
+// `Math.round(value) === value`
+const mathRoundSelector = [
+	'BinaryExpression',
+	methodCallSelector({method: 'round', object: 'Math', path: 'left'}),
+	equalsSelector,
+].join('');
+
+// `~~value === value`
+const bitwiseNotSelector = [
+	'BinaryExpression',
+	'[left.type="UnaryExpression"]',
+	'[left.operator="~"]',
+	'[left.argument.type="UnaryExpression"]',
+	'[left.argument.operator="~"]',
+	equalsSelector,
+].join('');
+
+function createNodeListener(sourceCode, variableGetter) {
+	return node => {
+		const variable = variableGetter(node);
+
+		if (!variable) {
+			return;
+		}
+
+		return {
+			node,
+			messageId: MESSAGE_ID,
+			data: {
+				variable,
+				original: sourceCode.getText(node),
+			},
+			/** @param {import('eslint').Rule.RuleFixer} fixer */
+			suggest: [{
+				messageId: MESSAGE_ID_SUGGEST,
+				fix: fixer => fixer.replaceText(node, `Number.isInteger(${variable})`),
+			}],
+		};
+	};
+}
+
+function getNodeName(node) {
+	switch (node.type) {
+		case 'Identifier': {
+			return node.name;
+		}
+
+		case 'ChainExpression': {
+			return getNodeName(node.expression);
+		}
+
+		case 'MemberExpression': {
+			return `${getNodeName(node.object)}.${getNodeName(node.property)}`;
+		}
+
+		default: {
+			return '';
+		}
```

Comment:
@sindresorhus what should we do here?

I've added the default case as to appease linting, but the calling code protects us from this branch being taken.

How should I approach appeasing Codecov?

Reply:
You could use an ignore comment:

```
/* c8 ignore next 3 */
```